test/LieGroupsTestSuite.jl/LieGroupsTestSuite.jl

```diff
@@ -0,0 +1,208 @@
+"""
+    LieGroupsTestSuite.jl
+
+This module provides tools and dummy structures to test functionality provided
+within `LieGroups.jl`.
+
+For every test function, several interactions to other functions can be activated.
+The following functions are expected to be available, since their defaults just pass through to the manifold
+* `is_point` both on the Lie group `G` and the Lie algebra `𝔤`
+* `isapprox(G,g,h)` and `issaprox(𝔤, X, Y)`
+* `copy(G,g)`
+* `norm(𝔤,X)`
+"""
+module LieGroupsTestSuite
+using LieGroups
+using Test
+
+#
+#
+# Dummy Types
+struct DummyOperation <: AbstractGroupOperation end
+struct DummySecondOperation <: AbstractGroupOperation end
+struct DummyManifold <: LieGroups.AbstractManifold{LieGroups.ℝ} end
+#
+#
+# test component functions
+
+"""
+    test_compose(G, g, h)
+
+Test functionality of `compose` for given Lie group elements `g`, `h`.
+
+# Keyword arguments
+
+* `test_inverse=true`: test that `g^{-1}g` is the identity (requires `inv`, `inv!`, and `is_identity`)
+* `test_identity=true`: test that composing with the identity yields the identity (requires `identity_element`)
+"""
+function test_compose(G::LieGroup, g, h; test_inverse=true, test_identity=true)
+    @testset "compose(G, g, h)" begin
+        k1 = compose(G, g, h)
+        k2 = copy(G, g)
+        compose!(G, k2, g, h)
+        @test isapprox(G, k1, k2)
+        if test_inverse
+            for g_ in [g, h]
+                g_inv = inv(G, g_)
+                k1 = compose(G, g_inv, g_)
+                compose!(G, k2, g_inv, g_)
+                @test isapprox(G, k1, k2)
+                @test is_identity(G, k1)
+                @test is_identity(G, k2)
+            end
+        end
+        if test_identity
+            for g_ in [g, h]
+                for e in [Identity(G), identity_element(G)]
+                    k1 = compose(G, g_, e)
+                    compose!(G, k2, g_, e)
+                    @test isapprox(G, k1, k2)
+                    k1 = compose(G, e, g_)
+                    compose!(G, k2, e, g_)
+                    @test isapprox(G, k1, k2)
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+"""
+    test_exp_log(G, g, h, X)
+
+Test functionality of `exp` and `log` for given Lie group elements `g`, `h` and
+a vector `X` from the Lie Algebra.
+
+!!! note
+    This function requires`Is_point(G, g)` and `is_point(𝔤, X)` to be implemented
```

Comment:
```suggestion
    This function requires `is_point(G, g)` and `is_point(LieAlgebra(G), X)` to be implemented
```
Maybe that would be more clear?

Reply:
sure.